File: python/socketd/socketd/transport/core/listener/SimpleListener.py
from socketd.transport.core.Listener import Listener
from socketd.transport.core.Message import Message
from socketd.transport.core.Session import Session


class SimpleListener(Listener):

    async def on_open(self, session: Session):
        pass

    async def on_message(self, session: Session, message: Message):
        pass

    async def on_close(self, session: Session):
        pass

    def on_error(self, session: Session, error):
        pass

File: javascript/socketd/src/socketd/transport/core/listener/SimpleListener.ts
import type {Session} from "../Session";
import type {Message} from "../Message";
import {Listener} from "../Listener";

/**
 * 简单监听器（一般用于占位）
 *
 * @author noear
 * @since 2.0
 */
export class SimpleListener implements Listener {
    onOpen(session: Session) {

    }

    onMessage(session: Session, message: Message) {

    }

    onReply(session: Session, message: Message) {

    }

    onSend(session: Session, message: Message) {
    }

    onClose(session: Session) {

    }

    onError(session: Session, error: Error) {

    }
}